In [1]:
import pandas as pd

## Train Data set

In [6]:
df = pd.read_excel("Data_Train.xlsx")
df.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

In [7]:
df.shape

(10683, 11)

## Test Data set

In [3]:
df1 = pd.read_excel("Test_set.xlsx")
df.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

In [8]:
df1.shape

(2671, 10)

### Let's merge them 

In [10]:
df = df.append(df1)
df.shape

C:\Users\seher\AppData\Local\Temp\ipykernel_812\3484042514.py:1: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(df1)


(13354, 11)

### Data set info 

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13354 entries, 0 to 2670
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Airline          13354 non-null  object 
 1   Date_of_Journey  13354 non-null  object 
 2   Source           13354 non-null  object 
 3   Destination      13354 non-null  object 
 4   Route            13353 non-null  object 
 5   Dep_Time         13354 non-null  object 
 6   Arrival_Time     13354 non-null  object 
 7   Duration         13354 non-null  object 
 8   Total_Stops      13353 non-null  object 
 9   Additional_Info  13354 non-null  object 
 10  Price            10683 non-null  float64
dtypes: float64(1), object(10)
memory usage: 1.2+ MB


## We have only 1 float column and all others are objects 
we need to convert them into suitable form

## Check for Null values

In [13]:
df.isnull().sum()

Airline               0
Date_of_Journey       0
Source                0
Destination           0
Route                 1
Dep_Time              0
Arrival_Time          0
Duration              0
Total_Stops           1
Additional_Info       0
Price              2671
dtype: int64

## here we got only 1 nan values in 2 columns and 2671 null bcz of test data set 

### let's have a look towards each columns 

In [16]:
df['Airline'].unique()

array(['IndiGo', 'Air India', 'Jet Airways', 'SpiceJet',
       'Multiple carriers', 'GoAir', 'Vistara', 'Air Asia',
       'Vistara Premium economy', 'Jet Airways Business',
       'Multiple carriers Premium economy', 'Trujet'], dtype=object)

In [17]:
len(df['Airline'].unique())

12

### we have 12 different categories, here we can use one hot encoding or label encoding as the airlines are not related to each other so the better approach is to use one hot encoding 

In [23]:
dummies = pd.get_dummies(df['Airline'], drop_first = True)
dummies.head(6)

Air India  GoAir  IndiGo  Jet Airways  Jet Airways Business  \
0          0      0       1            0                     0   
1          1      0       0            0                     0   
2          0      0       0            1                     0   
3          0      0       1            0                     0   
4          0      0       1            0                     0   
5          0      0       0            0                     0   

   Multiple carriers  Multiple carriers Premium economy  SpiceJet  Trujet  \
0                  0                                  0         0       0   
1                  0                                  0         0       0   
2                  0                                  0         0       0   
3                  0                                  0         0       0   
4                  0                                  0         0       0   
5                  0                                  0         1       0   

   Vistara  Vistara Premium economy  
0        0                        0  
1        0                        0  
2        0                        0  
3        0                        0  
4        0                        0  
5        0                        0

In [20]:
dummies.shape

(13354, 11)

## Let's Combine them with our df

In [24]:
df = pd.concat([df,dummies], axis = 1)

In [25]:
df.head(4)

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  ...  GoAir  \
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info  ...      0   
1    05:50         13:15   7h 25m     2 stops         No info  ...      0   
2    09:25  04:25 10 Jun      19h     2 stops         No info  ...      0   
3    18:05         23:30   5h 25m      1 stop         No info  ...      0   

   IndiGo  Jet Airways  Jet Airways Business  Multiple carriers  \
0       1            0                     0                  0   
1       0            0                     0                  0   
2       0            1                     0                  0   
3       1            0                     0                  0   

   Multiple carriers Premium economy  SpiceJet  Trujet  Vistara  \
0                                  0         0       0        0   
1                                  0         0       0        0   
2                                  0         0       0        0   
3                                  0         0       0        0   

   Vistara Premium economy  
0                        0  
1                        0  
2                        0  
3                        0  

[4 rows x 22 columns]

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13354 entries, 0 to 2670
Data columns (total 22 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Airline                            13354 non-null  object 
 1   Date_of_Journey                    13354 non-null  object 
 2   Source                             13354 non-null  object 
 3   Destination                        13354 non-null  object 
 4   Route                              13353 non-null  object 
 5   Dep_Time                           13354 non-null  object 
 6   Arrival_Time                       13354 non-null  object 
 7   Duration                           13354 non-null  object 
 8   Total_Stops                        13353 non-null  object 
 9   Additional_Info                    13354 non-null  object 
 10  Price                              10683 non-null  float64
 11  Air India                          13354 non-null  uint

### Drop Airline Column now

In [29]:
df.drop('Airline', axis = 1, inplace = True)

## Move to the second column 

In [30]:
df['Date_of_Journey']

0       24/03/2019
1        1/05/2019
2        9/06/2019
3       12/05/2019
4       01/03/2019
           ...    
2666     6/06/2019
2667    27/03/2019
2668     6/03/2019
2669     6/03/2019
2670    15/06/2019
Name: Date_of_Journey, Length: 13354, dtype: object

## Let's bring it into suitable formate and seperate them all in month, year and date columns

In [35]:
df['Date_of_Journey'].str.split('/').str[0]

0       24
1        1
2        9
3       12
4       01
        ..
2666     6
2667    27
2668     6
2669     6
2670    15
Name: Date_of_Journey, Length: 13354, dtype: object

In [36]:
df['Date'] = df['Date_of_Journey'].str.split('/').str[0]
df['Month'] = df['Date_of_Journey'].str.split('/').str[1]
df['Year'] = df['Date_of_Journey'].str.split('/').str[2]

In [39]:
df.head(1)

Date_of_Journey    Source Destination      Route Dep_Time  Arrival_Time  \
0      24/03/2019  Banglore   New Delhi  BLR → DEL    22:20  01:10 22 Mar   

  Duration Total_Stops Additional_Info   Price  ...  Jet Airways Business  \
0   2h 50m    non-stop         No info  3897.0  ...                     0   

   Multiple carriers  Multiple carriers Premium economy  SpiceJet  Trujet  \
0                  0                                  0         0       0   

   Vistara  Vistara Premium economy  Date  Month  Year  
0        0                        0    24     03  2019  

[1 rows x 24 columns]

In [40]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13354 entries, 0 to 2670
Data columns (total 24 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Date_of_Journey                    13354 non-null  object 
 1   Source                             13354 non-null  object 
 2   Destination                        13354 non-null  object 
 3   Route                              13353 non-null  object 
 4   Dep_Time                           13354 non-null  object 
 5   Arrival_Time                       13354 non-null  object 
 6   Duration                           13354 non-null  object 
 7   Total_Stops                        13353 non-null  object 
 8   Additional_Info                    13354 non-null  object 
 9   Price                              10683 non-null  float64
 10  Air India                          13354 non-null  uint8  
 11  GoAir                              13354 non-null  uint

### Change the type of newly created df columns from object to int 

In [44]:
df['Date'] = df['Date'].astype('int')
df['Month'] = df['Month'].astype('int')
df['Year'] = df['Year'].astype('int')

In [45]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13354 entries, 0 to 2670
Data columns (total 24 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Date_of_Journey                    13354 non-null  object 
 1   Source                             13354 non-null  object 
 2   Destination                        13354 non-null  object 
 3   Route                              13353 non-null  object 
 4   Dep_Time                           13354 non-null  object 
 5   Arrival_Time                       13354 non-null  object 
 6   Duration                           13354 non-null  object 
 7   Total_Stops                        13353 non-null  object 
 8   Additional_Info                    13354 non-null  object 
 9   Price                              10683 non-null  float64
 10  Air India                          13354 non-null  uint8  
 11  GoAir                              13354 non-null  uint

## Let's drop this column too

In [46]:
df.drop('Date_of_Journey', axis = 1, inplace = True)

## Checking the source Column 

In [47]:
df['Source'].unique()

array(['Banglore', 'Kolkata', 'Delhi', 'Chennai', 'Mumbai'], dtype=object)

### Again repeat one hot encoding as we did with Airline column

In [50]:
dummies = pd.get_dummies(df['Source'], drop_first = True)
df = pd.concat([df,dummies], axis = 1)
df.drop('Source', axis = 1, inplace = True)

In [51]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13354 entries, 0 to 2670
Data columns (total 26 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Destination                        13354 non-null  object 
 1   Route                              13353 non-null  object 
 2   Dep_Time                           13354 non-null  object 
 3   Arrival_Time                       13354 non-null  object 
 4   Duration                           13354 non-null  object 
 5   Total_Stops                        13353 non-null  object 
 6   Additional_Info                    13354 non-null  object 
 7   Price                              10683 non-null  float64
 8   Air India                          13354 non-null  uint8  
 9   GoAir                              13354 non-null  uint8  
 10  IndiGo                             13354 non-null  uint8  
 11  Jet Airways                        13354 non-null  uint

## Destination

In [52]:
df['Destination'].unique()

array(['New Delhi', 'Banglore', 'Cochin', 'Kolkata', 'Delhi', 'Hyderabad'],
      dtype=object)

In [53]:
dummies = pd.get_dummies(df['Destination'], drop_first = True)
df = pd.concat([df,dummies], axis = 1)
df.drop('Destination', axis = 1, inplace = True)

In [54]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13354 entries, 0 to 2670
Data columns (total 30 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Route                              13353 non-null  object 
 1   Dep_Time                           13354 non-null  object 
 2   Arrival_Time                       13354 non-null  object 
 3   Duration                           13354 non-null  object 
 4   Total_Stops                        13353 non-null  object 
 5   Additional_Info                    13354 non-null  object 
 6   Price                              10683 non-null  float64
 7   Air India                          13354 non-null  uint8  
 8   GoAir                              13354 non-null  uint8  
 9   IndiGo                             13354 non-null  uint8  
 10  Jet Airways                        13354 non-null  uint8  
 11  Jet Airways Business               13354 non-null  uint

# Route Column 

In [55]:
df['Route']

0                   BLR → DEL
1       CCU → IXR → BBI → BLR
2       DEL → LKO → BOM → COK
3             CCU → NAG → BLR
4             BLR → NAG → DEL
                ...          
2666          CCU → DEL → BLR
2667                CCU → BLR
2668          DEL → BOM → COK
2669          DEL → BOM → COK
2670          DEL → BOM → COK
Name: Route, Length: 13354, dtype: object

## As we have number of the stops so we don't need this info as much so we can drop it 

In [56]:
df.drop('Route', axis = 1 , inplace = True)

## Dep_Time column

In [57]:
df['Dep_Time']

0       22:20
1       05:50
2       09:25
3       18:05
4       16:50
        ...  
2666    20:30
2667    14:20
2668    21:50
2669    04:00
2670    04:55
Name: Dep_Time, Length: 13354, dtype: object

### This is in the form of hour and min time 

In [61]:
df['Dep_Time_Hour'] = df['Dep_Time'].str.split(':').str[0]
df['Dep_Time_Min'] = df['Dep_Time'].str.split(':').str[1]

In [62]:
df.head(4)

Dep_Time  Arrival_Time Duration Total_Stops Additional_Info    Price  \
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897.0   
1    05:50         13:15   7h 25m     2 stops         No info   7662.0   
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882.0   
3    18:05         23:30   5h 25m      1 stop         No info   6218.0   

   Air India  GoAir  IndiGo  Jet Airways  ...  Delhi  Kolkata  Mumbai  Cochin  \
0          0      0       1            0  ...      0        0       0       0   
1          1      0       0            0  ...      0        1       0       0   
2          0      0       0            1  ...      1        0       0       1   
3          0      0       1            0  ...      0        1       0       0   

   Delhi  Hyderabad  Kolkata  New Delhi  Dep_Time_Hour  Dep_Time_Min  
0      0          0        0          1             22            20  
1      0          0        0          0             05            50  
2      0          0        0          0             09            25  
3      0          0        0          0             18            05  

[4 rows x 31 columns]

In [63]:
df.drop('Dep_Time', axis = 1, inplace = True)

In [64]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13354 entries, 0 to 2670
Data columns (total 30 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Arrival_Time                       13354 non-null  object 
 1   Duration                           13354 non-null  object 
 2   Total_Stops                        13353 non-null  object 
 3   Additional_Info                    13354 non-null  object 
 4   Price                              10683 non-null  float64
 5   Air India                          13354 non-null  uint8  
 6   GoAir                              13354 non-null  uint8  
 7   IndiGo                             13354 non-null  uint8  
 8   Jet Airways                        13354 non-null  uint8  
 9   Jet Airways Business               13354 non-null  uint8  
 10  Multiple carriers                  13354 non-null  uint8  
 11  Multiple carriers Premium economy  13354 non-null  uint

## Arrival_Time 

In [67]:
df['Arrival_Time']

0       01:10 22 Mar
1              13:15
2       04:25 10 Jun
3              23:30
4              21:35
            ...     
2666    20:25 07 Jun
2667           16:55
2668    04:25 07 Mar
2669           19:15
2670           19:15
Name: Arrival_Time, Length: 13354, dtype: object

In [68]:
df['Arrival_Time'] = df['Arrival_Time'].apply( lambda x : x.split(' ')[0])

In [69]:
df['Arrival_Time']

0       01:10
1       13:15
2       04:25
3       23:30
4       21:35
        ...  
2666    20:25
2667    16:55
2668    04:25
2669    19:15
2670    19:15
Name: Arrival_Time, Length: 13354, dtype: object

In [70]:
df['Arrival_Time_hour'] = df['Arrival_Time'].str.split(':').str[0]
df['Arrival_Time_min'] = df['Arrival_Time'].str.split(':').str[1]

In [71]:
df.drop('Arrival_Time', axis = 1, inplace = True)

In [72]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13354 entries, 0 to 2670
Data columns (total 31 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Duration                           13354 non-null  object 
 1   Total_Stops                        13353 non-null  object 
 2   Additional_Info                    13354 non-null  object 
 3   Price                              10683 non-null  float64
 4   Air India                          13354 non-null  uint8  
 5   GoAir                              13354 non-null  uint8  
 6   IndiGo                             13354 non-null  uint8  
 7   Jet Airways                        13354 non-null  uint8  
 8   Jet Airways Business               13354 non-null  uint8  
 9   Multiple carriers                  13354 non-null  uint8  
 10  Multiple carriers Premium economy  13354 non-null  uint8  
 11  SpiceJet                           13354 non-null  uint

In [74]:
df['Duration'].head()

0    2h 50m
1    7h 25m
2       19h
3    5h 25m
4    4h 45m
Name: Duration, dtype: object

In [86]:
df['Duration_Hour'] = df['Duration'].str.split(' ').str[0].str.split('h').str[0]
df['Duration_Min'] = df['Duration'].str.split(' ').str[1].str.split('m').str[0]

In [89]:
df['Duration_Hour'].unique()

array(['2', '7', '19', '5', '4', '15', '21', '25', '13', '12', '26', '22',
       '23', '20', '10', '6', '11', '8', '16', '3', '27', '1', '14', '9',
       '18', '17', '24', '30', '28', '29', '37', '34', '38', '35', '36',
       '47', '33', '32', '31', '42', '39', '5m', '41', '40'], dtype=object)

In [92]:
df[df['Duration_Hour'] == '5m']

Duration Total_Stops Additional_Info    Price  Air India  GoAir  IndiGo  \
6474       5m     2 stops         No info  17327.0          1      0       0   
2660       5m     2 stops         No info      NaN          1      0       0   

      Jet Airways  Jet Airways Business  Multiple carriers  ...  Delhi  \
6474            0                     0                  0  ...      0   
2660            0                     0                  0  ...      0   

      Hyderabad  Kolkata  New Delhi  Dep_Time_Hour  Dep_Time_Min  \
6474          1        0          0             16            50   
2660          1        0          0             16            50   

      Arrival_Time_hour  Arrival_Time_min  Duration_Hour  Duration_Min  
6474                 16                55             5m           NaN  
2660                 16                55             5m           NaN  

[2 rows x 33 columns]

In [93]:
df.drop(6474, axis = 0, inplace = True)
df.drop(2660, axis = 0, inplace = True)

In [94]:
df['Duration_Hour'].unique()

array(['2', '7', '19', '5', '4', '15', '21', '25', '13', '12', '26', '22',
       '23', '20', '10', '6', '11', '8', '16', '3', '27', '1', '14', '9',
       '18', '17', '24', '30', '28', '29', '37', '34', '38', '35', '36',
       '47', '33', '32', '31', '42', '39', '41', '40'], dtype=object)

### Here we got nan value 

In [96]:
df['Duration_Min'].unique()

array(['50', '25', nan, '45', '30', '5', '15', '35', '10', '20', '55',
       '40'], dtype=object)

In [101]:
df['Duration_Min'] = df['Duration_Min'].fillna('0')

In [103]:
df['Duration_Min'].isnull().sum()

0

## Let's Convert the type of these variables into int

In [104]:
df['Duration_Hour'] = df['Duration_Hour'].astype('int')
df['Duration_Min'] = df['Duration_Min'].astype('int')

# Now let's combine them into the duration 
### the scale will be using is the minutes

In [107]:
df['Duration'] = df['Duration_Hour']*60 + df['Duration_Min']

In [109]:
df.head()

Duration Total_Stops Additional_Info    Price  Air India  GoAir  IndiGo  \
0       170    non-stop         No info   3897.0          0      0       1   
1       445     2 stops         No info   7662.0          1      0       0   
2      1140     2 stops         No info  13882.0          0      0       0   
3       325      1 stop         No info   6218.0          0      0       1   
4       285      1 stop         No info  13302.0          0      0       1   

   Jet Airways  Jet Airways Business  Multiple carriers  ...  Delhi  \
0            0                     0                  0  ...      0   
1            0                     0                  0  ...      0   
2            1                     0                  0  ...      0   
3            0                     0                  0  ...      0   
4            0                     0                  0  ...      0   

   Hyderabad  Kolkata  New Delhi  Dep_Time_Hour  Dep_Time_Min  \
0          0        0          1             22            20   
1          0        0          0             05            50   
2          0        0          0             09            25   
3          0        0          0             18            05   
4          0        0          1             16            50   

   Arrival_Time_hour  Arrival_Time_min  Duration_Hour  Duration_Min  
0                 01                10              2            50  
1                 13                15              7            25  
2                 04                25             19             0  
3                 23                30              5            25  
4                 21                35              4            45  

[5 rows x 33 columns]

### Now we don't need duration hour and min so let's drop it 

In [110]:
df.drop(['Duration_Hour','Duration_Min'], axis = 1, inplace = True)

## Total Stop Columns 

In [115]:
df['Total_Stops'].unique()

array(['non-stop', '2 stops', '1 stop', '3 stops', nan, '4 stops'],
      dtype=object)

In [119]:
df['Total_Stops'] = df['Total_Stops'].fillna('2')

In [120]:
df['Total_Stops'].isnull().sum()

0

In [122]:
df['Total_Stops'] = df['Total_Stops'].map({'non-stop':0,'1 stop':1,'2 stops':2,'3 stops':3,'4 stops':4, '2':2})

In [124]:
df['Total_Stops'].head()

0    0
1    2
2    2
3    1
4    1
Name: Total_Stops, dtype: int64

# Additional_Info

In [126]:
df['Additional_Info'].unique()

array(['No info', 'In-flight meal not included',
       'No check-in baggage included', '1 Short layover', 'No Info',
       '1 Long layover', 'Change airports', 'Business class',
       'Red-eye flight', '2 Long layover'], dtype=object)

In [127]:
dummies = pd.get_dummies(df['Additional_Info'], drop_first = True)
df = pd.concat([df,dummies], axis = 1)
df.drop('Additional_Info', axis = 1, inplace = True)

In [128]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13351 entries, 0 to 2670
Data columns (total 39 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Duration                           13351 non-null  int32  
 1   Total_Stops                        13351 non-null  int64  
 2   Price                              10681 non-null  float64
 3   Air India                          13351 non-null  uint8  
 4   GoAir                              13351 non-null  uint8  
 5   IndiGo                             13351 non-null  uint8  
 6   Jet Airways                        13351 non-null  uint8  
 7   Jet Airways Business               13351 non-null  uint8  
 8   Multiple carriers                  13351 non-null  uint8  
 9   Multiple carriers Premium economy  13351 non-null  uint8  
 10  SpiceJet                           13351 non-null  uint8  
 11  Trujet                             13351 non-null  uint

In [131]:
df['Dep_Time_Hour'] = df['Dep_Time_Hour'].astype('int')
df['Dep_Time_Min'] = df['Dep_Time_Min'].astype('int')
df['Arrival_Time_hour'] = df['Arrival_Time_hour'].astype('int')
df['Arrival_Time_min'] = df['Arrival_Time_min'].astype('int')

In [132]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13351 entries, 0 to 2670
Data columns (total 39 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Duration                           13351 non-null  int32  
 1   Total_Stops                        13351 non-null  int64  
 2   Price                              10681 non-null  float64
 3   Air India                          13351 non-null  uint8  
 4   GoAir                              13351 non-null  uint8  
 5   IndiGo                             13351 non-null  uint8  
 6   Jet Airways                        13351 non-null  uint8  
 7   Jet Airways Business               13351 non-null  uint8  
 8   Multiple carriers                  13351 non-null  uint8  
 9   Multiple carriers Premium economy  13351 non-null  uint8  
 10  SpiceJet                           13351 non-null  uint8  
 11  Trujet                             13351 non-null  uint

## Change the type of the coloumns 

In [153]:
def convert_to_int(col):
    if col.dtype == 'uint8':
        return col.astype(int)
    return col


df = df.apply(convert_to_int)

In [154]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13351 entries, 0 to 2670
Data columns (total 39 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Duration                           13351 non-null  int32  
 1   Total_Stops                        13351 non-null  int64  
 2   Price                              10681 non-null  float64
 3   Air India                          13351 non-null  int32  
 4   GoAir                              13351 non-null  int32  
 5   IndiGo                             13351 non-null  int32  
 6   Jet Airways                        13351 non-null  int32  
 7   Jet Airways Business               13351 non-null  int32  
 8   Multiple carriers                  13351 non-null  int32  
 9   Multiple carriers Premium economy  13351 non-null  int32  
 10  SpiceJet                           13351 non-null  int32  
 11  Trujet                             13351 non-null  int3

In [149]:
str(df['No info'].dtype)

'uint8'